src/solve.jl

```diff
@@ -102,7 +102,10 @@ function DiffEqBase.__init(prob::Union{DiffEqBase.AbstractODEProblem,DiffEqBase.
                                      prob.f.mass_matrix != I &&
                                      !(typeof(prob.f.mass_matrix)<:Tuple) &&
                                      ArrayInterface.issingular(prob.f.mass_matrix))
-
+  if alg isa AutoSwitch
+    alg.count = 0
```

Comment:
this is still not thread safe since if multiple threads are working then a separate thread can reset the count. We should make the algorithm type be a `struct` instead of a `mutable struct` to enforce correctness here, and have a `switching_cache` in the integrator for this.

Reply:
Could we just call `copy` here?

Reply:
we could copy the alg, but it would be so much nicer if it had immutable semantics like every other one. I think this and the StochasticDiffEq autoswitch are the only two that are mutable?

Reply:
Yeah